### ISSUE TITLE
enhanced ontology

### ISSUE THREAD

**Issue by touretzkyds** [touretzkyds/KnowledgeGraphDemo#58]
I've finished fleshing out the enhanced ontology for taxonomic relations. It's described in the wiki page at https://github.com/touretzkyds/KnowledgeGraphDemo/wiki/Ontology-design-notes.

Let's go ahead and implement this. The new properties can be visualized in the taxonomic hierarchy browser that Deya will be working on.

**Comment by deyaliao** [comment]
_PROPERTY_
A couple of clarifications for the properties:
- must we include "rdf:type rdf:Property", if the first line is an indication of the type itself? For example, this is the original declaration for taxon rank in the ontology file:
```
kgo:taxonRank a rdf:Property ;
        rdfs:comment      "The subject has taxonomic rank of the object"@en ;
	kgo:wikidataProperty	"P105"	;
        rdfs:domain       rdfs:Class ;
        rdfs:isDefinedBy  boltz:ontology ;
        rdfs:label        "taxonRank"@en ;
        rdfs:range        kgo:TaxonRank .
```
- should we change rdfs:comment to skos:definition? Everything in the [ontology file](https://github.com/touretzkyds/KnowledgeGraphServer/blob/dev/devel/Ontology.ttl) uses comment instead of definition. 

_CLASSES_
I was a bit confused on how to implement the classes in the Ontology file. These are the triples for the class, taxonomic rank in the current ontology file. 
```
kgo:TaxonRank a  rdfs:Class ;
        rdfs:comment      "The Class of all Taxonomic Ranks"@en ;
        rdfs:isDefinedBy  boltz:ontology ;
        rdfs:label        "TaxonRank"@en ;
        rdfs:subClassOf   rdfs:Class .
```

Do you want me to change the format, so it resembles something like
```
boltz:Q427626 a  rdfs:Class ;
<--> extra information
```
Additionally, when you say add all the other ranks in addition to genus under the "Triples needed" section, is it really just the triples already present in the [Data file](https://github.com/touretzkyds/KnowledgeGraphServer/blob/a2f3d98e91ddb44addfaed5f1dc1d3c2acd1af43/devel/Data.ttl#L409)? Are all these taxonomic ranks technically a class (you defined Genus as a class in the "Classes needed" section in the ontology design notes).

**Comment by touretzkyds** [comment]
1. You're correct that we don't have to add an rdf:type triple if we're declaring the concept using "a".
2. Yes, change rdfs:comment to skos:definition, as we're already using skos:definition for all the taxa and taxon ranks.
3. Yes, we should replace kgo:TaxonRank with boltz:Q427626. We should use boltz:Q-concepts wherever possible. But we're going to stick with kgo: for property names like kgo:taxonRank.
4. Yes, we already have the other taxonomic ranks in the data file, and you've already converted them to Q-concepts and added kgo:nextHigherRank and kgo:nextLowerRank properties. So we're almost done. But we're going to enhance all of them with additional relations; at the moment this is just kgo:instanceOf.

**Comment by deyaliao** [comment]
I've changed the [ontology file](https://github.com/touretzkyds/KnowledgeGraphServer/blob/dev/devel/Ontology.ttl) to reflect this changes. Please let me know if this is sufficient or what to change before I make a change to the server.

**Comment by touretzkyds** [comment]
This looks good. Just a few tweaks:
- kgo:Taxon should be replaced by boltz:Q16521
- kgo:taxonName should have wikidataProperty "P225"
- kgo:typeSpecies should have wikidataProperty "P427"

There are spacing/indentation issues when viewing the file in Github, e.g., see line 538 under kgo:taxonRank. Please make sure the file contains no tabs, only spaces. Also, on the concept deinition lines the string " a " should contain exactly one space before and after the "a".

**Comment by deyaliao** [comment]
I think I fixed these changes. Is it good now?

**Comment by touretzkyds** [comment]
On lines 13 and 18, "complementary" is misspelled as "complementry".

Other than that, it's good enought for now. There are still lots of places with weird spacing, but we can fix those later. They aren't hurting anything. So go ahead and deploy it. We'll see if it breaks anything in demo2d.

**Comment by deyaliao** [comment]
I fixed the spelling. I've also deployed it, and I think because we switched kgo:Taxon to be a q-boltz concept, the places that reference that don't work. For example (under "type", which is supposed to be "Taxon" I believe?): 
<img width="605" alt="Screen Shot 2022-07-07 at 7 10 02 PM" src="https://user-images.githubusercontent.com/61635230/177902596-1a2dc5a1-9273-4f63-909b-429a2a61adbe.png">

**Comment by touretzkyds** [comment]
Yes, all these taxa that have rdf:type kgo:Taxon need to change that to rdf:type boltz:Q16521. And then demo2d needs to be adjusted accordingly.
